### ISSUE TITLE
Changes of 'Number of PMS items to show..' not updating when doing "Refresh playlists/nodes"

### ISSUE THREAD

**Issue by dettwild** [croneter/PlexKodiConnect#203]
Perhaps I was expecting too much, but noticed that changing the Value under "Number of PMS items to show in widgets" does not have an effect.
Initially I was hoping that such a change would be picked up automatically, but figured that at least manually regenerating the Video Nodes (via the 'Refresh playlists/nodes' menu in PKC) would pass the new limit number to the video nodes. However this is NOT the case and only a full restart of Kodi does the trick.

I even have a suspicion as to why that is the case. It seems that the videonode generation uses the limit value from a `window('fetch_pms_item_number')` and that value seems to be assigned from the XML settings in service.py. I am still smuggling with reading the code (especially the sequence in which it runs), but could it be that this window value assignment only happens at boot time? If so I guess a simple solution would be to re-assign the respective window value from settings just before updating video nodes.

Or could it possibly be even more efficient to not even add the limit to the video node configs but instead dynamically assign it for each method call? I supposed that would happen in entrypoint? I would be delighted to quickly hear some pros and cons on the 2 options, but most important is of course the net functionality.

**Comment by croneter** [comment]
The current solution is easiest. Dynamic setting of the limit does not really work (or it already happens - e.g. your pull request ;-)). 

Works here. If you change the limit and exit the PKC settings, a Plex pop-up will appear telling you that Videonodes and playlist are being refreshed. Also, if debug logging is enabled:
```
14:28:49.297 T:8056  NOTICE: PLEX.kodimonitor -> DEBUG:: PKC settings changed: fetch_pms_item_number is now 1
14:28:49.300 T:8056  NOTICE: PLEX.kodimonitor -> Requesting playlist/nodes refresh
```
Be more specific - what is not working? Are you **within** the PKC video node, then navigate into a library? Is it On Deck or recently added? 
When you tie the video nodes to widgets, THEN change the limit, you might need to restart Kodi.

**Comment by dettwild** [comment]
Hmm, that is odd. 
What I did was changing the setting in the PKC Plugin and then go back to Kodi Movies and/or TV Show library to see the number of widget posters change. When that did not happen I went to kook at the video-node config files in the userdata-library folders to see if they already had the new limit, but they did not and only a restart of Kodi triggered the update. 
Doing the same again just now (shortly after a restart to keep the log-file short, just in case) I DID indeed see the mentioned message on screen (and later in the log-file), but only after first seeing two popups that database was being compressed. Even the widgets were updated shortly after, so this does work like you said (of course it would ;-).
Perhaps my issue earlier today was related to speed? What if a full background sync was running exactly when I change the settings? My RPi2 is sometimes really slow, even when just navigating on screen - not sure if it is my USB stick (SD card only for OS, Storage share with .kodi on the USB stick).
Should you have some tips on how to speed it up I would be more than happy to hear them.
But that is an entirely different story and I am therefore closing this issue here.
